core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlProtocolUnmarshaller.java

```diff
@@ -135,6 +163,34 @@ private boolean isExplicitPayloadMember(SdkField<?> field) {
         return field.containsTrait(PayloadTrait.class);
     }
 
+    private boolean hasExplicitBlobPayloadMember(SdkPojo sdkPojo) {
+        return sdkPojo.sdkFields()
+                      .stream()
+                      .anyMatch(f -> isExplicitPayloadMember(f) && f.marshallingType() == MarshallingType.SDK_BYTES);
+    }
+
+    private boolean hasExplicitStringPayloadMember(SdkPojo sdkPojo) {
+        return sdkPojo.sdkFields()
+                      .stream()
+                      .anyMatch(f -> isExplicitPayloadMember(f) && f.marshallingType() == MarshallingType.STRING);
+    }
+
+    private boolean hasPayloadMembersOnUnmarshall(SdkPojo sdkPojo) {
+        return sdkPojo.sdkFields()
+                      .stream()
+                      .anyMatch(f -> f.location() == MarshallLocation.PAYLOAD || locationInUri(f.location()));
+    }
+
+    private static boolean locationInUri(MarshallLocation location) {
+        switch (location) {
+            case PATH:
+            case QUERY_PARAM:
+                return true;
+            default:
+                return false;
+        }
+    }
```

Comment:
nit: `locationInUri` as a method name sounds like it'd return a location within the URI where the parameters are defined. Maybe `isInUri(location)`?

Reply:
+1, will update here and in `JsonProtocolUnmarshaller`